### PR TITLE
don't update current withdrawn amount when rate limit is reset when duration hasn't passed

### PR DESCRIPTION
**Pull request by viraj124**
- closes https://github.com/FuelLabs/fuel-bridge/issues/288

### CODE REVIEW

File: packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal/v3/FuelMessagePortalV3.sol [hunk 1]
```diff
@@ -79,31 +79,21 @@ contract FuelMessagePortalV3 is FuelMessagePortalV2 {
      * @notice Resets the rate limit amount.
      * @param _amount The amount to reset the limit to.
      */
-    function resetRateLimitAmount(uint256 _amount) external onlyRole(SET_RATE_LIMITER_ROLE) {
-        uint256 withdrawalLimitAmountToSet;
-        bool amountWithdrawnLoweredToLimit;
-        bool withdrawalAmountResetToZero;
-        
+    function resetRateLimitAmount(uint256 _amount) external onlyRole(SET_RATE_LIMITER_ROLE) {   
+        // currentPeriodAmount is not updated when duration hasn't passed to avoid a edge case scenario where extra amount can be withdrawn from the bridge when it sshoulddn't be
```
Comment (PraneshASP) [inline packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal/v3/FuelMessagePortalV3.sol:83]: typo -> "sshoulddn't"

File: packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal/v3/FuelMessagePortalV3.sol [hunk 2]
```diff
@@ -79,31 +79,21 @@ contract FuelMessagePortalV3 is FuelMessagePortalV2 {
...
+        // fo instance if rate limit is reduced & made less than the current withddrawn amount & increased to orginal before the duration ends then extra amount can be withdrawn
```
Comment (PraneshASP) [inline packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal/v3/FuelMessagePortalV3.sol:84]: typo -> "fo"

File: packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal/v3/FuelMessagePortalV3.sol [hunk 3]
```diff
@@ -79,31 +79,21 @@ contract FuelMessagePortalV3 is FuelMessagePortalV2 {
...
+        // if rate limit is reduced is reduced & made less then the current withdrawn amount then the withdraw operation from the bridge would revert until the limit is increasedd or the duration has passed
```
Comment (PraneshASP) [inline packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal/v3/FuelMessagePortalV3.sol:85]: typo -> "increasedd"

Comment (PraneshASP) [inline packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal/v3/FuelMessagePortalV3.sol:85]: typo: "is reduced is reduced"

File: packages/solidity-contracts/contracts/messaging/gateway/FuelERC20Gateway/FuelERC20GatewayV4.sol [hunk 4]
```diff
@@ -153,11 +153,10 @@ contract FuelERC20GatewayV4 is
      * @param _amount The amount to reset the limit to.
      * @param _rateLimitDuration The new rate limit duration.
      */
-    function resetRateLimitAmount(address _token, uint256 _amount, uint256 _rateLimitDuration) external onlyRole(SET_RATE_LIMITER_ROLE) {
-        uint256 withdrawalLimitAmountToSet;
-        bool amountWithdrawnLoweredToLimit;
-        bool withdrawalAmountResetToZero;
-        
+    function resetRateLimitAmount(address _token, uint256 _amount, uint256 _rateLimitDuration) external onlyRole(SET_RATE_LIMITER_ROLE) {   
+        // currentPeriodAmount is not updated when duration hasn't passed to avoid a edge case scenario where extra amount can be withdrawn from the bridge when it sshoulddn't be
```
Comment (PraneshASP) [inline packages/solidity-contracts/contracts/messaging/gateway/FuelERC20Gateway/FuelERC20GatewayV4.sol:157]: ^ same typos as above

File: packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal/v3/FuelMessagePortalV3.sol [hunk 5]
```diff
@@ -79,31 +79,21 @@ contract FuelMessagePortalV3 is FuelMessagePortalV2 {
...
             }
-            withdrawalAmountResetToZero = true;
-        } else {
-            // If the withdrawn amount is higher, it is set to the new limit amount
-            if (_amount < currentPeriodAmount) {
-                withdrawalLimitAmountToSet = _amount;
-                amountWithdrawnLoweredToLimit = true;
-            }
+
+            currentPeriodAmount = 0;
```
Comment (PraneshASP) [inline packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal/v3/FuelMessagePortalV3.sol:92]: can we also add a test scenario to make sure that `currentPeriodAmount` is set to zero as expected.

**Review by PraneshASP — CHANGES_REQUESTED** [review]
few nits

File: packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal/v3/FuelMessagePortalV3.sol [hunk 6]
```diff
@@ -79,31 +79,21 @@ contract FuelMessagePortalV3 is FuelMessagePortalV2 {
      * @notice Resets the rate limit amount.
      * @param _amount The amount to reset the limit to.
      */
-    function resetRateLimitAmount(uint256 _amount) external onlyRole(SET_RATE_LIMITER_ROLE) {
-        uint256 withdrawalLimitAmountToSet;
-        bool amountWithdrawnLoweredToLimit;
-        bool withdrawalAmountResetToZero;
-        
+    function resetRateLimitAmount(uint256 _amount) external onlyRole(SET_RATE_LIMITER_ROLE) {   
+        // currentPeriodAmount is not updated when duration hasn't passed to avoid a edge case scenario where extra amount can be withdrawn from the bridge when it shouldn't be
```
Comment (DefiCake) [inline packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal/v3/FuelMessagePortalV3.sol:83]: Instead of having these long comments, please add a new section in `ARCHITECTURE.md`:

- Let's give credits to the original authors of this (Linea bridge), put links to the original implementation. Do this in the code as well, following natspec
- In the markdown, highlight the differences between Linea and our implementation.
- Describe an scenario and the outcomes, using both Linea 's and our implementation.

File: packages/integration-tests/tests/bridge_erc20.ts [hunk 7]
```diff
@@ -359,6 +359,7 @@ describe('Bridging ERC20 tokens', async function () {
 
   describe('Bridge ERC20 from Fuel', async () => {
     const NUM_TOKENS = 10000000000000000000n;
+    const largeRateLimit = `30`;
```
Comment (DefiCake) [inline packages/integration-tests/tests/bridge_erc20.ts:362]: I see integration tests, but are there unit tests?

Comment (DefiCake) [reply packages/integration-tests/tests/bridge_erc20.ts:362]: Unless we are referring to different things, they do: https://github.com/FuelLabs/fuel-bridge/blob/6030a40ce9c58a533c09f73e837f85ab4784ef58/packages/solidity-contracts/test/behaviors/erc20GatewayV4.behavior.test.ts#L935

File: docs/ARCHITECTURE.md [hunk 8]
```diff
@@ -407,3 +491,7 @@ As it can be derived from the diagrams above, there are entities performing miss
...
 - Block committer: The block committer links the activity in the Fuel blockchain by pushing state updates back to Ethereum, where users can use these updates to prove certain aspects of the Fuel blockchain activity (such as withdrawals). If a block committer uploads rogue data to Ethereum, it can enable fraudulent behaviour (for example, uploading withdrawals that never happened). New states uploaded by the committer are timelocked (i.e. they cannot be used to prove L2 activity). If rogue states remain undetected for a time longer than the "finalization window", then a malicious actor can finally propagate the nefarious activity to Ethereum and ultimately extract funds out of our contracts. At that point, the loss is considered final and other avenues must be pursued for the recovery.
+
+```
```
Comment (PraneshASP) [inline docs/ARCHITECTURE.md:495]: do we need this?

**Review by PraneshASP — DISMISSED** [review]

**Review by PraneshASP — APPROVED** [review]

**Review by DefiCake — APPROVED** [review]
